generator/internal/round_robin_decorator_generator.cc

```diff
@@ -372,9 +372,10 @@ future<Status>
   CcPrint(R"""(
 std::shared_ptr<$stub_class_name$>
 $round_robin_class_name$::Child() {
-  std::lock_guard<std::mutex> lk(mu_);
-  auto child = children_[current_];
+  std::unique_lock<std::mutex> lk(mu_);
+  auto const& child = children_[current_];
```

Comment:
And perhaps `auto const current = current_;` and then `return children_[current];` would move some more arithmetic outside the lock.

Reply:
I think those would make no difference with modern compilers:

https://godbolt.org/z/vaa5Yq3hr

But if you think they can make the intent better, I would be happy to use that.